File: 백준/Gold/1918. 후위 표기식/후위 표기식.py
infix = list(input())

st = []
result = []
for i in infix:
    if i.isalpha():
        result.append(i)
    else:
        if i == '(':
            st.append(i)
        elif i == '+' or i == '-':
            while st and st[-1] != '(':
                result.append(st.pop())
            st.append(i)
        elif i == '*' or i == '/':
            while st and (st[-1] == '*' or st[-1] == '/'):
                result.append(st.pop())
            st.append(i)
        else:
            while st and st[-1] != '(':
                result.append(st.pop())
            st.pop()

while st:
    result.append(st.pop())
    
for i in result:
    print(i, end = '')

File: 백준/Gold/1918. 후위 표기식/후위 표기식.cc
#include <bits/stdc++.h>
using namespace std;

int main() {
    string input;
    cin >> input;

    vector<char> op;
    for (char c: input) {
        if (isalpha(c)) {
            cout << c;
            continue;
        }
        if (c == '(')
            op.push_back(c);
        else if (c == ')') {
            while (!op.empty() && op.back() != '(') {
                cout << op.back();
                op.pop_back();
            }
            op.pop_back();
        } else if (c == '*' || c == '/') {
            while (!op.empty() && (op.back() == '*' || op.back() == '/')){
                cout << op.back();
                op.pop_back();
            }
            op.push_back(c);
        } else if (c == '+' || c == '-') {
            while (!op.empty() && op.back() != '(') {
                cout << op.back();
                op.pop_back();
            }
            op.push_back(c);
        }

    }

    while (!op.empty()) {
        cout << op.back();
        op.pop_back();
    }

    return 0;
}